lib/membrane_webm/parser.ex

```diff
@@ -0,0 +1,276 @@
+    end
+  end
+
+  def parse(<<type::unsigned-integer-size(8)>>, :uint, :FlagInterlaced) do
+    case type do
+      # Unknown status.This value SHOULD be avoided.
+      0 -> :undetermined
+      # Interlaced frames.
+      1 -> :interlaced
+      # No interlacing.
+      2 -> :progressive
+    end
+  end
+
+  def parse(<<type::unsigned-integer-size(8)>>, :uint, :ChromaSitingHorz) do
+    case type do
+      0 -> :unspecified
+      1 -> :left_collocated
+      2 -> :half
+    end
+  end
+
+  def parse(<<type::unsigned-integer-size(8)>>, :uint, :ChromaSitingVert) do
+    case type do
+      0 -> :unspecified
+      1 -> :top_collocated
+      2 -> :half
+    end
+  end
+
+  # per RFC https://datatracker.ietf.org/doc/html/rfc8794#section-7.1
+  def parse(<<>>, :integer, _name) do
+    0
+  end
+
+  def parse(bytes, :integer, _name) do
+    s = byte_size(bytes) * 8
+    <<num::signed-big-integer-size(s)>> = bytes
+    num
+  end
+
+  # per RFC https://datatracker.ietf.org/doc/html/rfc8794#section-7.2
+  def parse(<<>>, :uint, _name) do
+    0
+  end
+
+  def parse(bytes, :uint, _name) do
+    :binary.decode_unsigned(bytes, :big)
+  end
+
+  # per RFC https://datatracker.ietf.org/doc/html/rfc8794#section-7.3
+  def parse(<<>>, :float, _name) do
+    0
+  end
+
+  def parse(<<num::float-big>>, :float, _name) do
+    num
+  end
+
+  def parse(bytes, :string, :CodecID) do
+    codec_string = Enum.join(for <<c::utf8 <- bytes>>, do: <<c::utf8>>)
+
+    case codec_string do
+      "A_OPUS" -> :opus
+      "A_VORBIS" -> :vorbis
+      "V_VP8" -> :vp8
+      "V_VP9" -> :vp9
+    end
+  end
+
+  def parse(bytes, :string, _name) do
+    chars = for <<c::utf8 <- bytes>>, do: <<c::utf8>>
+    chars |> Enum.take_while(fn c -> c != <<0>> end) |> Enum.join()
+  end
+
+  def parse(bytes, :utf_8, _name) do
+    bytes
+    |> String.codepoints()
+    |> Enum.reduce("", fn codepoint, result ->
+      <<parsed::8>> = codepoint
+      if parsed == 0, do: result, else: result <> <<parsed>>
+    end)
+  end
+
+  # per RFC https://datatracker.ietf.org/doc/html/rfc8794#section-7.6
+  def parse(<<>>, :date, _name) do
+    {{2001, 1, 1}, {0, 0, 0}}
+  end
+
+  def parse(<<nanoseconds::big-signed>>, :date, _name) do
+    seconds_zero = :calendar.datetime_to_gregorian_seconds({{2001, 1, 1}, {0, 0, 0}})
+    seconds = div(nanoseconds, Time.nanosecond()) + seconds_zero
+    :calendar.gregorian_seconds_to_datetime(seconds)
+  end
+
+  def parse(bytes, :binary, :SimpleBlock) do
+    # https://tools.ietf.org/id/draft-lhomme-cellar-matroska-04.html#rfc.section.6.2.4.4
+
+    # track_number is a vint with size 1 or 2 bytes
+    %{vint: track_number_vint, rest: body} = Vint.parse(bytes)
+    <<timecode::integer-signed-size(16), flags::bitstring-size(8), data::binary>> = body
+    <<keyframe::1, reserved::3, invisible::1, lacing::2, discardable::1>> = flags
+
+    if reserved != 0 do
+      IO.warn("SimpleBlock reserved bits in header flag should all be 0 but they are #{reserved}")
```

Comment:
Again, `Membrane.Logger.warn` and again ask yourself a question if you are able to continue operating while this incident happens.